### PR TITLE
GetHistory() and its helper methods are moved from storage to container

### PR DESCRIPTION
**Pull request by rgayon**

### CODE REVIEW

File: tests.py [hunk 1]
```diff
@@ -76,7 +76,7 @@ def testParseArguments(self):
 
   def testDetectStorageFail(self):
     """Tests that the DockerExplorer.DetectStorage function fails on
-    non-existing Docker directory."""
+    Docker directory."""
```
Comment (tomchop) [inline tests.py]: You removed something that seemed correct?

Comment (rgayon) [reply tests.py]: I blame Git =(

File: docker_explorer/lib/container.py [hunk 2]
```diff
@@ -92,9 +94,113 @@ def __init__(self, docker_directory, container_id, docker_version=2):
...
+      with open(layer_info_path) as layer_info_file:
+        layer_info = json.load(layer_info_file)
+        return layer_info
+    return None
+
+  def GetOrderedLayers(self):
+    """Returns an array of the sorted image ID for a container.
+
+    Returns:
+      list(str): a list of layer IDs (hashes).
```
[the remark this answers is not in the document]
Comment (rgayon) [reply docker_explorer/lib/container.py]: done

File: docker_explorer/lib/container.py [hunk 3]
```diff
@@ -92,9 +94,113 @@ def __init__(self, docker_directory, container_id, docker_version=2):
...
+          'content', hash_method, layer_id)
+    layer_info = None
+    if os.path.isfile(layer_info_path):
+      with open(layer_info_path) as layer_info_file:
+        layer_info = json.load(layer_info_file)
+        return layer_info
+    return None
+
+  def GetOrderedLayers(self):
+    """Returns an array of the sorted image ID for a container.
```
[the remark this answers is not in the document]
Comment (rgayon) [reply docker_explorer/lib/container.py]: done

File: docker_explorer/lib/container.py [hunk 4]
```diff
@@ -92,9 +94,113 @@ def __init__(self, docker_directory, container_id, docker_version=2):
...
+    """
+    if self.docker_version == 1:
+      layer_info_path = os.path.join(
+          self.docker_directory, 'graph', layer_id, 'json')
+    elif self.docker_version == 2:
+      hash_method, layer_id = layer_id.split(':')
+      layer_info_path = os.path.join(
+          self.docker_directory, 'image', self.storage_driver, 'imagedb',
+          'content', hash_method, layer_id)
+    layer_info = None
```
[the remark this answers is not in the document]
Comment (rgayon) [reply docker_explorer/lib/container.py]: done

File: docker_explorer/lib/container.py [hunk 5]
```diff
@@ -92,9 +94,113 @@ def __init__(self, docker_directory, container_id, docker_version=2):
...
+    if os.path.isfile(layer_info_path):
+      with open(layer_info_path) as layer_info_file:
+        layer_info = json.load(layer_info_file)
```
[the remark this answers is not in the document]
Comment (rgayon) [reply docker_explorer/lib/container.py]: done

File: docker_explorer/lib/container.py [hunk 6]
```diff
@@ -92,9 +94,113 @@ def __init__(self, docker_directory, container_id, docker_version=2):
...
+                          layer_id, 'layersize')
+      size = int(open(path).read())
+    # TODO(romaing) Add docker storage v2 support
+    return size
+
+  def GetLayerInfo(self, layer_id):
+    """Gets a docker FS layer information.
+
+    Returns:
+      dict: the container information.
```
[the remark this answers is not in the document]
Comment (rgayon) [reply docker_explorer/lib/container.py]: done

File: docker_explorer/lib/container.py [hunk 7]
```diff
@@ -92,9 +94,113 @@ def __init__(self, docker_directory, container_id, docker_version=2):
...
+
+    Returns:
+      int: the size of the layer in bytes.
+    """
+    size = 0
+    if self.docker_version == 1:
+      path = os.path.join(self.docker_directory, 'graph',
+                          layer_id, 'layersize')
+      size = int(open(path).read())
+    # TODO(romaing) Add docker storage v2 support
```
[the remark this answers is not in the document]
Comment (rgayon) [reply docker_explorer/lib/container.py]: I did a bit or research, and it looks like currently the container history seem broken on "v2 and not AuFS" storage.
Filed #28 to keep track of this =(

File: docker_explorer/lib/container.py [hunk 8]
```diff
@@ -92,9 +94,113 @@ def __init__(self, docker_directory, container_id, docker_version=2):
...
           self.docker_directory, 'image', self.storage_driver, 'layerdb',
           'mounts', container_id)
       with open(os.path.join(c_path, 'mount-id')) as mount_id_file:
         self.mount_id = mount_id_file.read()
+
+  def GetLayerSize(self, layer_id):
+    """Returns the size of the layer.
+
+    Args:
+      layer_id (str): the layer id to get the size of.
```
[the remark this answers is not in the document]
Comment (rgayon) [reply docker_explorer/lib/container.py]: done

File: docker_explorer/lib/container.py [hunk 9]
```diff
@@ -53,8 +54,9 @@ def __init__(self, docker_directory, container_id, docker_version=2):
       errors.BadContainerException: if there was an error when parsing
         container_info_json_path
     """
+    self.docker_version = docker_version
```
[the remark this answers is not in the document]
Comment (rgayon) [reply docker_explorer/lib/container.py]: done

**Pull request merged**